safe_control_gym/envs/gym_control/cartpole.py

```diff
@@ -205,7 +205,7 @@ def __init__(self,
             self.X_GOAL = np.hstack([self.TASK_INFO['stabilization_goal'][0], 0., 0., 0.])  # x = {x, x_dot, theta, theta_dot}.
         elif self.TASK == Task.TRAJ_TRACKING:
             POS_REF, VEL_REF, _ = self._generate_trajectory(traj_type=self.TASK_INFO['trajectory_type'],
-                                              traj_length=self.EPISODE_LEN_SEC,
+                                              traj_length=self.EPISODE_LEN_SEC + self.CTRL_TIMESTEP, # Added to increment the trajectory by 1 (PR #119)
```

Comment:
Instead increase generate trajectory by ```times = np.arange(0, traj_length + sample_time, sample_time)```. Double check that this is one longer than before.

Reply:
Yes, I think the point is that for an episode with 1 second, 2Hz control frequency, we want "1 reset, 2 control steps" and the reference should be of size 3.

So I would fully encapsulate the mismatch by 1 between the reference and the control steps in the reference generation method.

Reply:
Added.